### PR TITLE
Fixed inheritance and `RoleField` definition on `Role` model mixins.

### PR DESCRIPTION
**Pull request by jathanism**
# Closes: #3291 
# What's Changed

- Updated `RoleModelMixin` to make `RoleField` null and blank explicitly
- Updated `RoleRequiredRoleMixin` base to inherit from `models.Model` and not subclass `RoleModelMixin`

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/models/roles.py [hunk 1]
```diff
@@ -47,13 +47,13 @@ class RoleModelMixin(models.Model):
...
 
-    role = RoleField()
+    role = RoleField(null=True, blank=True)
 
     class Meta:
         abstract = True
 
 
-class RoleRequiredRoleModelMixin(RoleModelMixin):
+class RoleRequiredRoleModelMixin(models.Model):
```
Comment (glennmatthews) [inline nautobot/extras/models/roles.py:56]: Why was this problematic, for my education?

File: nautobot/extras/models/roles.py [hunk 2]
```diff
@@ -47,13 +47,13 @@ class RoleModelMixin(models.Model):
     Abstract base class for any model which may have roles.
     """
 
-    role = RoleField()
+    role = RoleField(null=True, blank=True)
```
Comment (timizuoebideri1) [inline nautobot/extras/models/roles.py:50]:
```suggestion
    role = RoleField()
```
`blank` and `null` are set to be `True` by default, Why setting it here again?

Comment (jathanism) [reply nautobot/extras/models/roles.py:50]: I was just being explicit. I think perhaps fixing the inheritance is enough to alleviate the reported issue.

File: nautobot/extras/models/roles.py (hunk 1 above)
Comment (jathanism) [reply nautobot/extras/models/roles.py:56]: Subclassing an abstract model to create another abstract model that is overloading the same field is apparently not doing what it was intended to do according to the bug report. I was able to confirm the somewhat confusing behavior.

**Review by timizuoebideri1 — APPROVED** [review]

**Pull request merged**